### PR TITLE
Mark a keyless list as operational

### PR DESCRIPTION
**Pull request by kot-begemot-uk**
The YANG RFCs mandate that keyless lists are operational only.

Signed-off-by: Anton Ivanov <anton.ivanov@cambridgegreys.com>

### CODE REVIEW

**Comment by jeff-yin** [comment]
@hpersh please provide review

File: configure.ac [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #                                               -*- Autoconf -*-
 # Process this file with autoconf to produce a configure script.
 AC_PREREQ([2.69])
-AC_INIT([opx-base-model], [3.151.0], [ops-dev@lists.openswitch.net])
+AC_INIT([opx-base-model], [3.152.0], [ops-dev@lists.openswitch.net])
```
Comment (rakeshdatta) [inline configure.ac]: package version in debian/changelog and confugure.ac should be same : '3.151.0+num1'

File: debian/changelog [hunk 2]
```diff
@@ -1,3 +1,9 @@
+opx-base-model (3.152.0+nmu1) unstable; urgency=medium
```
Comment (rakeshdatta) [inline debian/changelog]: package version in debian/changelog and confugure.ac should be same : '3.151.0+num1'

File: yang-models/dell-base-alms.yang [hunk 3]
```diff
@@ -22,12 +22,16 @@ module dell-base-alms {
 
   All rights reserved.";
 
+  revision 2018-09-21 {
```
Comment (rakeshdatta) [inline yang-models/dell-base-alms.yang]: We have taken care of most of the pyang errors internally. They should be available soon.
In this case, you should write the revision as:

 revision 2018-09-21 {
    description "Fixes for error from pyang validation and IETF compliance";
    reference "Network Platform Abstraction";
  }

Without the reference, pyang would through IETF compliance errors.

File: yang-models/dell-base-if-vlan.yang [hunk 4]
```diff
@@ -6,21 +6,22 @@ module dell-base-if-vlan {
...
     }
 
+    import iana-if-type { prefix "ianaift"; }
+
     organization "Dell EMC";
     contact "http://www.dell.com/support";
 
     description "This model augments the standard interface and adds Dell EMC specific
     				vlan extensions.";
+    revision "2018-09-21" {
```
Comment (rakeshdatta) [inline yang-models/dell-base-if-vlan.yang]: For every revision statement,  we have standardized the inclusion of a reference statement. See the example below.

 revision "2018-08-10" {
        description "Fixing pyang IETF errors and adding documentation to the model.";
        reference "Network Platform Abstraction";
    }

File: yang-models/dell-base-pas.yang [hunk 5]
```diff
@@ -14,18 +14,21 @@ module dell-base-pas {
 
   description "Configuration and operational data for OS10 Platform Abtraction Service";
 
-  revision 2015-04-03 {
-    description "Initial revision.";
+  revision 2018-09-21 {
```
Comment (rakeshdatta) [inline yang-models/dell-base-pas.yang]: revision "2018-08-10" {
        description "Fixing pyang IETF errors and adding documentation to the model.";
        reference "Network Platform Abstraction";
    }

These changes are redundant mostly. The modified files to be synced soon. I would suggest you can wait for these files to get the updated changes, and then re-check for pyang validity. Most of them are taken care of. If any remaining, you can fix them.

**Comment by kot-begemot-uk** [comment]
You mean 3.152.0+nmu1, right?

Will do shortly. Apologies, I did not notice that dch marked it as
"non-maintainer upload 1".

A.

File: configure.ac (hunk 1 above)
Comment (hpersh) [reply configure.ac]: Version should be 3.151.0+opx1

File: debian/changelog [hunk 6]
```diff
@@ -1,3 +1,9 @@
+opx-base-model (3.152.0) unstable; urgency=medium
```
Comment (hpersh) [inline debian/changelog]: Version should be 3.151.0+opx1

**Pull request closed without merging**